### PR TITLE
IOU metrics

### LINKED ISSUE #713: IoU Object Detection Metric
The IoU metric for object detection is an essential.  We need to offer this under `keras_cv.metrics`.

### PR DESCRIPTION
**Pull request by soma2000-lang**
fixes #713

## Before submitting

- [x]  Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you write any new necessary tests?
- [x] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

 @LukeWood @tanzhenyu @fchollet @lengzq @ianstenbit   I am new to this ecosystem.I would be great if you can review the  PR and suggest the changes required.

Regards.

### CODE REVIEW

**Comment by tanzhenyu** [comment]
Is this still required? We have completed the COCOCallback which gives pretty much everything.
(Sorry for not updating the original request more timely)

**Comment by soma2000-lang** [comment]
@tanzhenyu Sorry ,did not check .Closing this.
